runtime/src/local_runqueue_mts.c

```diff
@@ -0,0 +1,281 @@
+#include <stdint.h>
+
+#include "arch/context.h"
+#include "client_socket.h"
+#include "current_sandbox.h"
+#include "debuglog.h"
+#include "global_request_scheduler.h"
+#include "local_runqueue.h"
+#include "local_runqueue_mts.h"
+#include "panic.h"
+#include "priority_queue.h"
+#include "sandbox_functions.h"
+#include "runtime.h"
+
+__thread static struct priority_queue *local_runqueue_mts_guaranteed;
+__thread static struct priority_queue *local_runqueue_mts_default;
+
+extern __thread struct priority_queue *worker_thread_timeout_queue;
+
+/**
+ * Get Per-Worker-Module priority for Priority Queue ordering
+ * @param element the PWM
+ * @returns the priority of the head of the PWM
+ */
+static inline uint64_t
+perworker_module_get_priority(void *element)
+{
+	struct perworker_module_sandbox_queue *pwm     = (struct perworker_module_sandbox_queue *)element;
+	struct sandbox *                       sandbox = NULL;
+	priority_queue_top_nolock(pwm->sandboxes, (void **)&sandbox);
+	return (sandbox) ? sandbox->absolute_deadline : UINT64_MAX;
+}
+
+/**
+ * Checks if the run queue is empty
+ * @returns true if empty. false otherwise
+ */
+bool
+local_runqueue_mts_is_empty()
+{
+	return priority_queue_length_nolock(local_runqueue_mts_guaranteed) == 0
+	       && priority_queue_length_nolock(local_runqueue_mts_default) == 0;
+}
+
+/**
+ * Adds a sandbox to the run queue
+ * @param sandbox
+ * @returns pointer to sandbox added
+ */
+void
+local_runqueue_mts_add(struct sandbox *sandbox)
+{
+	assert(sandbox != NULL);
+
+	struct perworker_module_sandbox_queue *pwm               = &sandbox->module->pwm_sandboxes[worker_thread_idx];
```

Comment:
Want to validate that the `pwm_sandboxes` elements are cache-line aligned.

Reply:
I have this. But not sure how to validate it in this line, though. @gparmer 

```
struct perworker_module_sandbox_queue {
	struct priority_queue *  sandboxes;
	struct module *          module; 
	struct module_timeout    module_timeout;
	enum MULTI_TENANCY_CLASS mt_class; 
} __attribute__((aligned(128)));
```